p2p/conn_tracker.go

```diff
@@ -0,0 +1,75 @@
+package p2p
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+)
+
+type connectionTracker interface {
+	AddConn(net.IP) error
+	CheckConn(net.IP) (int, bool)
+	RemoveConn(net.IP)
+}
+
+type connTrackerImpl struct {
```

Comment:
It seems like this implementation should do the trick. Any reason you opted for a in-house implementation instead of a ready-to-use rate limier, e.g. https://github.com/sethvargo/go-limiter?

The above already garbage collects and maps directly to our API & needs. We can certainly keep this implementation, but I was just curious. Generally in favor of having to maintain less code :)

Reply:
This makes sense. I think it's pretty simple, and I don't see places where we're using go-limiter otherwise (and it looks like we'd need some kind of wrapper code/etc. anyway, so I'm not sure if it's a huge burden.)  I put the rate-limiting behind an interface so we could change it later if we needed.